tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift

```diff
@@ -233,37 +237,286 @@ internal final class WindowsRemoteProcess: RemoteProcess {
+    GetExitCodeThread(hThread, &threadExitCode)
+    if threadExitCode == 0 {
+      print("LoadLibraryW failed \(threadExitCode)")
+      return
+    }
+
+    // Unload the dll and deallocate the dll path from the remote process
+    if !unloadDllAndPathRemote(dwProcessId: dwProcessId, hKernel32: hKernel32,
+                               dllPathRemote: dllPathRemote) {
+      print("Failed to unload the remote dll")
+      return
+    }
+  }
+
+  private func allocateDllPathRemote() -> UnsafeMutableRawPointer? {
+    // The path to the dll assuming it's in the same directory as swift-inspect.
+    let swiftInspectPath = ProcessInfo.processInfo.arguments[0]
+    let dllPath: String =
+        URL(fileURLWithPath: swiftInspectPath).deletingLastPathComponent().path +
+        "\\SwiftInspectClient.dll"
+    var dllPathRemote = UnsafeMutableRawPointer(bitPattern: 0)
+    var dllPathUtf16Len: UInt = 0
+    var dllPathWritten = false
+    dllPath.withCString(encodedAs: UTF16.self) {
+      // Check that the dll file exists
+      let dwAttrib = GetFileAttributesW($0)
+      if dwAttrib == INVALID_FILE_ATTRIBUTES ||
+         Int32(dwAttrib) & FILE_ATTRIBUTE_DIRECTORY != 0 {
+        print("\(dllPath) doesn't exist")
+        return
+      }
+      // Allocate memory and write dllPath in the remote process
+      dllPathUtf16Len = UInt(wcslen($0) * MemoryLayout<WCHAR>.size)
+      dllPathRemote = VirtualAllocEx(self.process,
+                                     nil,
+                                     SIZE_T(dllPathUtf16Len + 1), // +1 for `\0`
+                                     DWORD(MEM_COMMIT),
+                                     DWORD(PAGE_READWRITE))
+      if dllPathRemote == LPVOID(bitPattern: 0) {
+        print("VirtualAllocEx failed \(GetLastError())")
+        return
+      }
+      if !WriteProcessMemory(self.process, dllPathRemote, $0,
+                             SIZE_T(dllPathUtf16Len), nil) {
+        print("WriteProcessMemory failed \(GetLastError())")
+        VirtualFreeEx(self.process, dllPathRemote, 0, DWORD(MEM_RELEASE))
+        return
+      }
+      dllPathWritten = true
+    }
+    if !dllPathWritten {
+      print("Failed to write dll path in the remote process")
+      return nil
+    }
+    return dllPathRemote
+  }
+
+  private func unloadDllAndPathRemote(
+      dwProcessId: DWORD, hKernel32: HMODULE!,
+      dllPathRemote: UnsafeMutableRawPointer) -> Bool {
+    // Get the dll module handle in the remote process to use it to
+    // unload it below.
+    // GetExitCodeThread returns a DWORD (32-bit) but the HMODULE
+    // returned from LoadLibraryW is a 64-bit pointer and may be truncated.
+    // So, search for it using the snapshot instead.
+    var hDllModule = HMODULE(bitPattern: 0)
+    let hModuleSnapshot: HANDLE =
+        CreateToolhelp32Snapshot(DWORD(TH32CS_SNAPMODULE), dwProcessId)
+    if hModuleSnapshot == INVALID_HANDLE_VALUE {
+      print("CreateToolhelp32Snapshot failed")
```

Comment:
This doesn't give us much context on what happened, can we get extended information?

Reply:
Done